File: Remove Nth Node From End of List.js
var removeNthFromEnd = function(head, n) {
    if(!head.next) return null;
    var current = head;
    var delay = head;
    
    while(current){
        if(n === -1) delay = delay.next;
        else n--;
        current = current.next;
    }
    if(delay === head & n === 0) head = delay.next;
    else delay.next = delay.next.next;
    return head;
};

File: Remove Nth Node From End of List.py
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        dummy = ListNode()
        dummy.next = head
        ahead_node = head

        while n:
            ahead_node = ahead_node.next
            n -= 1
        
        delay_node = dummy
        while ahead_node:
            ahead_node = ahead_node.next
            delay_node = delay_node.next
        
        delay_node.next = delay_node.next.next

        return dummy.next
